### ISSUE TITLE
RDLogmanager: On Event add/edit, reorder buttons segfault

### ISSUE THREAD

**Issue by alexolivan** [ElvishArtisan/rivendell#714]
Keep crushing those pesky qt5 bugs!!! here's another one:

While editing/creating ann event, If you try to re-order the post-import (I bet it may happen on pre-import too) carts using the up/down side buttons, rdlogmanager segfaults. I attach the output.
Apparently, a similar issue with Lists getting the classic out-of-bound except:

```
ASSERT failure in QList<T>::swap: "index out of range", file /usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h, line 738
Aborted
```

**Comment by ElvishArtisan** [comment]
Unable to reproduce the problem here. Any chance you could provide a 'step-by-step' procedure for goosing it?

**Comment by alexolivan** [comment]
OK... The bug is there, but today I was having trouble to find it too.

The issue comes mainly as you add events to an empty list and then you reorder them before saving the event.
It triggers both while adding and editing an event.
It happens both with post an pre import carts.
But it is tricky to find it while working normally.

Today but, during my tests, I have managed to find two different outputs during crashes, the one that I saw earlier, and the following one today: 
```
operador@debian:~$ rdlogmanager
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 3367, resource id: 8400529, major code: 40 (TranslateCoords), minor code: 0
ASSERT failure in QList<T>::swap: "index out of range", file /usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h, line 738
Aborted
```

**Comment by ElvishArtisan** [comment]
Thanks. I'll try some more here.

Have you tried running it under GDB? A backtrace would let us nail this in short order.

Also, what distro have you been testing on? (All of my attempts to date have been on CentOS 7).

**Comment by ElvishArtisan** [comment]
'Kay, I _think_ (heh, heh) that I've gotten to the bottom of this. More model-magic! Fixed in 3f114d3.

Please pound on this, as I've never actually been able to goose the original problem here.

**Comment by alexolivan** [comment]
Yep! Bug is crushed!
